src/lib/mongoBackend/collectionOperations.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+*
+* Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Fermín Galán
+*/
+
+#include "mongoBackend/MongoGlobal.h"
+#include "logMsg/traceLevels.h"
+
+#include "mongo/client/dbclient.h"
+
+using namespace mongo;
+
+/* ****************************************************************************
```

Comment:
I think there is value in grouping all the operations in the same module, as they are closely related in their purpose (and in its structure, e.g. all them start getting the connection, have same try/catch with 2 catch clauses). Even more, it could happen (we will see as new function are being added) that some common functionality between functions may be factorized.

Probably query() is not a good name for the function, as is very close to mongo driver query() method. Maybe doQuery() could be better...